src/braket/devices/local_simulator.py

```diff
@@ -134,13 +135,18 @@ def _run_internal(
 
 @_run_internal.register
 def _(circuit: Circuit, simulator: BraketSimulator, shots, *args, **kwargs):
-    if DeviceActionType.JAQCD not in simulator.properties.action:
-        raise NotImplementedError(f"{type(simulator)} does not support qubit gate-based programs")
-    validate_circuit_and_shots(circuit, shots)
-    program = circuit.to_ir()
-    qubits = circuit.qubit_count
-    results = simulator.run(program, qubits, shots, *args, **kwargs)
-    return GateModelQuantumTaskResult.from_object(results)
+    if DeviceActionType.OPENQASM in simulator.properties.action:
+        validate_circuit_and_shots(circuit, shots)
+        program = circuit.to_ir(ir_type=IRType.OPENQASM)
+        results = simulator.run(program, shots, *args, **kwargs)
+        return GateModelQuantumTaskResult.from_object(results)
+    elif DeviceActionType.JAQCD in simulator.properties.action:
+        validate_circuit_and_shots(circuit, shots)
+        program = circuit.to_ir(ir_type=IRType.OPENQASM)
```

Comment:
* Should this be IRType.JAQCD instead ?
* Can we add a test for this scenario ?

Reply:
Yes, good catch. Added.